core/services/workflows/syncer/workflow_registry.go

```diff
@@ -221,22 +263,31 @@ func (w *workflowRegistry) Start(_ context.Context) error {
 				return
 			}
 
-			w.lggr.Debugw("Loading initial workflows for DON", "DON", don.ID)
-			loadWorkflowsHead, err := w.loadWorkflows(ctx, don, reader)
-			if err != nil {
-				// TODO - this is a temporary fix to handle the case where the chainreader errors because the contract
-				// contains no workflows.  To track: https://smartcontract-it.atlassian.net/browse/CAPPL-393
-				if !strings.Contains(err.Error(), "attempting to unmarshal an empty string while arguments are expected") {
-					w.lggr.Errorw("failed to load workflows", "err", err)
-					return
-				}
+			// Start goroutines to gather changes from Workflow Registry contract
+			switch w.syncStrategy {
+			case SyncStrategyEvent:
+				go w.syncUsingEventStrategy(ctx, don, reader)
+			case SyncStrategyReconciliation:
+				w.syncUsingReconciliationStrategy(ctx, don, reader)
+			}
 
-				loadWorkflowsHead = &types.Head{
-					Height: "0",
+			// Handle events from the events channel
+			w.lggr.Debug("running handleEventLoop")
+			for {
+				select {
+				case <-ctx.Done():
+					w.lggr.Debug("shutting down handleEventLoop")
+					return
+				case event := <-w.eventCh:
+					if event == nil {
+						continue
```

Comment:
🤔 This shouldn't be necessary unless we're somehow sending nils on the channel, which we should really avoid